src/browser/cdp/selectivity/utils.ts

```diff
@@ -0,0 +1,210 @@
+    sourceMap.sourceRoot = sourceRoot || sourceMap.sourceRoot;
+
+    for (let i = 0; i < sourceMap.sources.length; i++) {
+        if (isWebpackProtocol(sourceMap.sources[i])) {
+            sourceMap.sources[i] = sourceMap.sources[i].slice(WEBPACK_PROTOCOL.length + 1);
+        }
+    }
+
+    return sourceMap;
+};
+
+export const extractSourceFilesDeps = async (
+    source: string,
+    sourceMaps: string,
+    startOffsets: number[],
+    sourceRoot: string,
+): Promise<Set<string>> => {
+    const dependantSourceFiles = new Set<string>();
+    const sourceMapsParsed = patchSourceMapSources(JSON.parse(sourceMaps), sourceRoot);
+
+    const consumer = (await new SourceMapConsumer(sourceMapsParsed)) as BasicSourceMapConsumer;
+
+    let sourceOffset = source.indexOf("\n");
+    const offsetToLine = [0];
+
+    while (sourceOffset !== -1) {
+        offsetToLine.push(++sourceOffset);
+        sourceOffset = source.indexOf("\n", sourceOffset);
+    }
+
+    for (const startOffset of startOffsets) {
+        let line = sortedIndex(offsetToLine, startOffset);
+
+        if (startOffset < offsetToLine[line]) {
+            line--;
+        }
+
+        const column = startOffset - offsetToLine[line];
+        const position = consumer.originalPositionFor({ line: line + 1, column });
+
+        if (position.source) {
+            dependantSourceFiles.add(position.source);
+        }
+    }
+
+    return dependantSourceFiles;
+};
+
+export const hasProtocol = (fileUrlLikePath: string): boolean => {
+    if (!fileUrlLikePath.includes("://")) {
+        return false;
+    }
+
+    try {
+        return Boolean(new URL(fileUrlLikePath).protocol);
+    } catch {
+        return false;
+    }
+};
+
+const ensurePosixRelativeDependencyPathExists = memoize((posixRelativePath: string): void => {
+    const relativePath = posixRelativePath.replaceAll(path.posix.sep, path.sep);
+
+    if (fs.existsSync(relativePath)) {
+        return;
+    }
+
+    throw new Error(`Selectivity: Couldn;t find "${relativePath}", which is test's dependency`);
+});
+
+/**
+ * @param cssDependencies SORTED uniq array of css dependenciy URI's
+ * @param jsDependencies SORTED uniq array of js dependenciy URI's
+ * @returns sorted uniq arrays of relative paths
+ */
+export const transformSourceDependencies = (
+    cssDependencies: string[],
+    jsDependencies: string[],
+): NormalizedDependencies => {
+    const nodeModulesLabel = "node_modules/";
+    const css: string[] = [];
+    const js: string[] = [];
+    const modules: string[] = [];
+
+    const classifyDependency = (dependency: string, typedResultArray: string[]): void => {
+        dependency = decodeURIComponent(softFileURLToPath(dependency));
+
+        if (hasProtocol(dependency)) {
+            throw new Error(`Selectivity: Found unsupported protocol in dependencies ("${dependency}")`);
+        }
+
+        const dependencyRelativePath = path.posix.relative(path.posix.resolve(), path.posix.resolve(dependency));
```

Comment:
- using "path.posix.resolve()" to get posix cwd
- "path.posix.resolve(dependency)" resolves absolute path from cwd